api/src/main/java/com/nicknathanjustin/streamercontracts/donations/OpenDonationDto.java

```diff
@@ -0,0 +1,30 @@
+package com.nicknathanjustin.streamercontracts.donations;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+
+import java.math.BigDecimal;
+import java.util.UUID;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class OpenDonationDto {
+    private UUID donationId;
+    private BigDecimal donationAmount;
+    private String description;
+    private String streamerName;
+
+    static OpenDonationDto fromEntity(@NonNull final DonationModel donationModel) {
+        return OpenDonationDto.builder()
```

Comment:
Is there an advantage to using the "builder" pattern here vs just making a constructor for the object that takes in the DonationModel as a parameter.

Reply:
I prefer builders for clarity (and i think most people most of the time should prefer them to constructors). Constructors you need to remember the order of params which can be annoying for a long list of params.